### PR TITLE
types for pomelo module

### PR DESCRIPTION
**Pull request by PengjuDuan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/pomelo/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline types/pomelo/tsconfig.json]: This must be `true` for all new packages.

File: types/pomelo/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+     */
+    export class Application {
+        /**
+         * Get application base path
+         *
+         *  // cwd: /home/game/
+         *  pomelo start
+         *  // app.getBase() -> /home/game
+         *
+         * @return {string} application base path
```
Comment (andy-ms) [inline types/pomelo/index.d.ts]: Don't add types in jsdoc comments, the code is already a type definition after all.

**Review by ghost — COMMENTED** [review]
What package is this for? `npm install pomelo` doesn't seem to include an `Application` class.

PR is missing files:
* `types/pomelo/pomelo-tests.ts` to test the type definitions. (i.e., import `pomelo` and use it in a typical way)
* `types/pomelo/tslint.json`, which should contain `{ "extends": "dtslint/dt.json" }`.

File: types/pomelo/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,890 @@
+declare module pomelo {
```
Comment (andy-ms) [inline types/pomelo/index.d.ts]: Needs a header. See `types/abs/index.d.ts` for an example.

**Comment by RyanCavanaugh** [comment]
@PengjuDuan Please fix the failures indicated in the Travis CI log.

**Comment by minestarks** [comment]
I'm closing this PR due to staleness. It has test failures and revision requests that haven't been addressed in two months. Please open a new PR if you'd still like to get the changes in. Thanks!

**Pull request closed without merging**